### DISCUSSION TITLE
End / Last slide partial for revealjs presentation

### DISCUSSION THREAD

**Discussion by royfrancis** [quarto-dev/quarto-cli#8084] [Feature Requests]
### Description

I think it would be nice to have an end slide attributes and end slide partial similar to [title slide partial](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/revealjs/pandoc/title-slide.html).  A final slide that would be automatically added to the end of all presentations with some predetermined content. Or different content if manually specified through metadata. Is this something I could create myself or is it too complex?

**Comment by mcanouil** [comment]
See <https://github.com/mcanouil/quarto-revealjs-coeos/blob/main/template.qmd>, I am using a shortcode that I just paste at the end `{{< include _extensions/mcanouil/coeos/about.qmd >}}`. There is not much more that is needed.

**Reply by royfrancis** [reply]
Nice! Exactly what I had in mind. That's straight forward. But, is it possible to set that as a default in _quarto.yml without having to add that last slide and shortcode at the bottom of every qmd file?

**Reply by mcanouil** [reply]
You still need to write the content somewhere ...

I don't think adding a single shortcode is much of an hassle and I personally don't make several slidedeck inside one project, so setting that in `_quarto.yml` rather than in the file itself does not seem an improvement.

You can write your own extension that does the addition post-render or in whatever way you like.

Personally, I don't think there's should be a partial for this as it would be very opinionated and setting this would mean users would have to opt-out every time.
So I strongly believe, that should be left to the users to define their own "last slide" as they see fit.
For instance, YAML key are not necessarily processed by Quarto so you loose features.
So really a lot of work on Quarto side for a little benefits and nearly no flexibility for the users.

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/985de311-78a9-4908-9bf2-b659383e9c0e">

**Reply by royfrancis** [reply]
Fair points! But, my use case is a course website where different teachers are contributing materials on their topic. They would only work on the their content while the end slide is automatically added by my theme template. The end slide will essentially be the same for all topics.

**Reply by cscheid** [reply]
> by my theme template
 
If you have a "theme template", you could provide that content by changing the actual Pandoc template, see https://quarto.org/docs/journals/templates.html#revealjs-partials.
